### PR TITLE
Workaround for pyasn1 0.4

### PR DESCRIPTION
**Pull request by tiran**
pyasn1 0.4 changed handling of ANY containers in a backwards
incompatible way. Try to decode ASN.1 octet streams explicitly for
pyasn1 0.3, then fall back to 0.4 behavior.

Tested with pyasn1 0.3.7 and 0.4.4.

Fixes: https://pagure.io/freeipa/issue/7685
Signed-off-by: Christian Heimes <cheimes@redhat.com>

### CODE REVIEW

**Comment by frasertweedale** [comment]
@tiran does anything need to be done on the encoding side (see `IPACertificate.__encode_extension`)?

**Comment by tiran** [comment]
@frasertweedale good catch! Of course it's also broken :rotating_light:

File: ipalib/x509.py [hunk 1]
```diff
@@ -285,7 +289,7 @@ def extended_key_usage_bytes(self):
             return None
 
         ekurfc = rfc2459.ExtKeyUsageSyntax()
-        for i, oid in enumerate(eku):
+        for i, oid in enumerate(sorted(eku)):
```
Comment (frasertweedale) [inline ipalib/x509.py]: I'm curious: what's the reason for this change?

Comment (tiran) [reply ipalib/x509.py]: EKU is a set of OIDs. Without sort, the sorting of EKU items depends on Python's hash randomziation. The change makes the sort order of EKUs stable. It simplifies testing and makes the output predictable.

**Comment by tiran** [comment]
master:

* e1a30d3ce32aef2492689de8eff09360f5480a99 Workaround for pyasn1 0.4

**Comment by frasertweedale** [comment]
On Wed, Sep 26, 2018 at 09:55:06PM -0700, Christian Heimes wrote:
> tiran commented on this pull request.
> 
> 
> 
> > @@ -285,7 +289,7 @@ def extended_key_usage_bytes(self):
>              return None
>  
>          ekurfc = rfc2459.ExtKeyUsageSyntax()
> -        for i, oid in enumerate(eku):
> +        for i, oid in enumerate(sorted(eku)):
> 
> EKU is a set of OIDs. Without sort, the sorting of EKU items depends on Python's hash randomziation. The change makes the sort order of EKUs stable. It simplifies testing and makes the output predictable.
>
Fair enough.
